bundler/default/lib/saito/storage.js

```diff
@@ -31,6 +31,56 @@ class Storage {
     }
   }
 
+  loadTransactions(type="all", num=50,  mycallback) {
+
+    let message = "archive";
+    let data = {};
+        data.request = "load";
+        data.type = type;
+        data.num = num;
+        data.publickey = this.app.wallet.returnPublicKey();
+
+    this.app.network.sendRequestWithCallback(message, data, function(obj) {
+      let txs = [];
+      if (obj) {
+        if (obj.txs) {
+          if (obj.txs.length > 0) {
+            txs = obj.txs.map(tx => new saito.transaction(JSON.parse(tx)));
+          }
+        }
+      }
+      mycallback(txs);
+    });
+
+  }
+
+  loadTransactionsByKeys(keys, type="all", num=50,  mycallback) {
```

Comment:
This duplicates most functionality of the above function. Could this be shared code moved out or optimized?